sw/device/tests/hmac_enc_irq_test.c

```diff
@@ -0,0 +1,147 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/hmac_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+
+dif_hmac_t hmac;
+dif_rv_plic_t plic;
+
+/**
+ * Flags to check whether the interrupts have fired.
+ */
+static bool hmac_empty_irq;
+static bool hmac_done_irq;
+
+const test_config_t kTestConfig;
+
+HMAC_TESTUTILS_ASSERT_LITTLE_ENDIAN;
+static const dif_hmac_transaction_t kHmacTransactionConfig = {
+    .digest_endianness = kDifHmacEndiannessLittle,
+    .message_endianness = kDifHmacEndiannessLittle,
+};
+
+/**
+ * Services HMAC interrupts.
+ *
+ * For the HMAC done the interrupt is cleared internally by `dif_hmac_finish`.
+ */
+static void handle_hmac_isr(const dif_rv_plic_irq_id_t interrupt_id) {
+  // NOTE: This initialization is superfluous, since the `default` case below
+  // is effectively noreturn, but the compiler is unable to prove this.
+  dif_hmac_irq_t hmac_irq = 0;
+
+  switch (interrupt_id) {
+    // Digest verification is done in a separate test, thus the digest is
+    // discarded.
+    case kTopEarlgreyPlicIrqIdHmacHmacDone:
+      hmac_irq = kDifHmacIrqHmacDone;
+      dif_hmac_digest_t dummy_digest;
+      dif_result_t res = dif_hmac_finish(&hmac, &dummy_digest);
+      CHECK(res == kDifOk, "HMAC finish failure = %d", res);
```

Comment:
There's a valid argument [here](https://github.com/lowRISC/opentitan/pull/10911#discussion_r810158627) that we should avoid these checkings inside the irq handler. Which may be a problem when running with preemption.

Reply:
We might want to implement `irq_testutils` that follow what you have done for otbn (keeping irq_id and peripheral_id as static variables). These would be set in the interrupt service routine, and the basic IRQ handling such as claim/acknowledge handled in the ISR, with all the rest of the logic outside the ISR.

Reply:
Just an FYI, there are autogenerated IRQ testutils in: https://github.com/lowRISC/opentitan/blob/master/sw/device/lib/testing/autogen/isr_testutils.h

Reply:
Done.